### PR TITLE
Added ReaderServiceImplTest test.

### PR DESCRIPTION
**Pull request by sormanzje**

### CODE REVIEW

File: bundle/service/src/test/java/hu/uni/miskolc/iit/swtest/team3/service/ReaderServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    public void testListBooks() {
+        when(testBookDao.read()).thenReturn(testBookList);
+
+        Assert.assertEquals(testBookList, readerServiceImpl.listBooks());
+        verify(testBookDao).read();
+    }
+
+    @Test(expected = UnsuccessfulOperationException.class)
+    public void listBooksException(){
+        when(testBookDao.read()).thenThrow((new UnsuccessfulOperationException("Could not get the list of books!")));
```
Comment (djavorek) [inline bundle/service/src/test/java/hu/uni/miskolc/iit/swtest/team3/service/ReaderServiceImplTest.java]: Dao needs to throw DataAccessException, that is catched by service and throws the Unsuccessful..Exception

Comment (aronnagy) [reply bundle/service/src/test/java/hu/uni/miskolc/iit/swtest/team3/service/ReaderServiceImplTest.java]: Thanks for the advice! Now It is fixed and works well. (For more info check the next commit).

**Comment by djavorek** [comment]
It still looks like there are uncovered lines :|

File: bundle/service/src/test/java/hu/uni/miskolc/iit/swtest/team3/service/ReaderServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    public void testListBooks() {
+        when(testBookDao.read()).thenReturn(testBookList);
+
+        Assert.assertEquals(testBookList, readerServiceImpl.listBooks());
+        verify(testBookDao).read();
+    }
+
+    @Test(expected = UnsuccessfulOperationException.class)
+    public void testListBooksException() {
+        Mockito.when(testBookDao.read()).thenThrow(Mockito.mock(DataAccessException.class)).thenReturn(testBookList);
```
Comment (djavorek) [inline bundle/service/src/test/java/hu/uni/miskolc/iit/swtest/team3/service/ReaderServiceImplTest.java]: Is .thenReturn neccessary?

File: bundle/service/src/test/java/hu/uni/miskolc/iit/swtest/team3/service/ReaderServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    public void testListBorrowings() {
+        when(testBorrowingDao.readByUser(testUser)).thenReturn(testBorrowingList);
+
+        Assert.assertEquals(testBorrowingList, readerServiceImpl.listBorrowings(testUser));
+        verify(testBorrowingDao).readByUser(testUser);
+    }
+
+    @Test(expected = UnsuccessfulOperationException.class)
+    public void testListBorrowingsException() {
+        Mockito.when(testBorrowingDao.readByUser(testUser)).thenThrow(Mockito.mock(DataAccessException.class)).thenReturn(testBorrowingList);
```
Comment (djavorek) [inline bundle/service/src/test/java/hu/uni/miskolc/iit/swtest/team3/service/ReaderServiceImplTest.java]: Is .thenReturn neccessary?

File: bundle/service/src/test/java/hu/uni/miskolc/iit/swtest/team3/service/ReaderServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+    @Test
+    public void testRequestBook() {
+        when(testBookDao.read(testBook.getIsbn())).thenReturn(testBook);
+        readerServiceImpl.requestBook(testBook, testUser);
+    }
+
+    @Test(expected = UnsuccessfulOperationException.class)
+    public void testRequestBookException() {
+        Mockito.when(testBookDao.read(testBook.getIsbn())).thenThrow(Mockito.mock(DataAccessException.class)).thenReturn(testBook);
```
Comment (djavorek) [inline bundle/service/src/test/java/hu/uni/miskolc/iit/swtest/team3/service/ReaderServiceImplTest.java]: Is .thenReturn neccessary?

File: bundle/service/src/test/java/hu/uni/miskolc/iit/swtest/team3/service/ReaderServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+        readerServiceImpl.requestBook(testBook, testUser);
+    }
+    
+
+    @Test
+    public void testCheckAvailability(){
+        when(testBookDao.read(testBook.getIsbn())).thenReturn(testBook);
+        
+        if(testBook.getAvailableCopies() > 0){
```
Comment (djavorek) [inline bundle/service/src/test/java/hu/uni/miskolc/iit/swtest/team3/service/ReaderServiceImplTest.java:134]: How does this work?
How many available copies the "testBook" has
If you want to check both if and else you should do it in separate test or something.
I dont think using if is a good idea here

File: bundle/service/src/test/java/hu/uni/miskolc/iit/swtest/team3/service/ReaderServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+        else {
+            Assert.assertFalse(readerServiceImpl.checkAvailability(testBook));
+        }
+    }
+
+    @Test(expected = UnsuccessfulOperationException.class)
+    public void testCheckAvailabilityException() {
+        Mockito.when(testBookDao.read(testBook.getIsbn())).thenThrow(Mockito.mock(DataAccessException.class)).thenReturn(testBook);
+
+        if(testBook.getAvailableCopies() > 0){
```
Comment (djavorek) [inline bundle/service/src/test/java/hu/uni/miskolc/iit/swtest/team3/service/ReaderServiceImplTest.java:146]: Which one do you test?
If or else it wont check the two in the same time

**Review by djavorek — CHANGES_REQUESTED** [review]

**Review by djavorek — APPROVED** [review]

**Pull request merged**